src/Content/Text/BBCode.php

```diff
@@ -413,16 +334,16 @@ public static function getAttachedData($body, $item = [])
 					}
 				}
 			} elseif (preg_match_all("(\[img\](.*?)\[\/img\])ism", $body, $pictures, PREG_SET_ORDER)) {
-				if ((count($pictures) == 1) && !$has_title) {
+				if ((count($pictures) > 0) && !$has_title) {
 					$post['type'] = 'photo';
-					$post['image'] = $pictures[0][1];
-					$post['text'] = str_replace($pictures[0][0], '', $body);
 				} elseif (count($pictures) > 0) {
 					$post['type'] = 'link';
 					$post['url'] = $plink;
+				}
+
+				if (count($pictures) > 0) {
```

Comment:
The `count($pictures) > 0` conditions in this block are superfluous because `preg_match_all()` already returns the count of full matches, which is the same as `count($pictures) > 0`. If there aren't any pictures, PHP will not get into this `elseif` block at all.

Suggested simplification:
```php
			} elseif (preg_match_all("(\[img\](.*?)\[\/img\])ism", $body, $pictures, PREG_SET_ORDER)) {
				if ($has_title) {
					$post['type'] = 'link';
					$post['url'] = $plink;
				} else {
					$post['type'] = 'photo';
					$post['image'] = $pictures[0][1];
					$post['text'] = str_replace($pictures[0][0], '', $body);
				}

				$post['image'] = $pictures[0][1];'
				$post['text'] = $body;

				foreach ($pictures as $picture) {
					$post['text'] = trim(str_replace($picture[0], '', $post['text']));
				}
			}
```

Reply:
Changed as well, only with the difference that "image" and "text" don't need to be filled in the `else` case.